pkg/server/plugin/datastore/sql/sql_test.go

```diff
@@ -2045,5 +2047,2468 @@ func (s *PluginSuite) TestConfigure() {
+		{
+			dialect: "mysql",
+			by:      []string{"parent-id", "selector-subset-many"},
+			query: `
+SELECT
+	E.id AS e_id,
+	E.entry_id AS entry_id,
+	E.spiffe_id,
+	E.parent_id,
+	E.ttl AS reg_ttl,
+	E.admin,
+	E.downstream,
+	E.expiry,
+	S.id AS selector_id,
+	S.type AS selector_type,
+	S.value AS selector_value,
+	B.trust_domain,
+	D.id AS dns_name_id,
+	D.value AS dns_name
+FROM
+	registered_entries E
+LEFT JOIN
+	(SELECT 1 AS joinItem UNION SELECT 2 UNION SELECT 3) AS joinItems ON TRUE
+LEFT JOIN
+	selectors S ON joinItem=1 AND E.id=S.registered_entry_id
+LEFT JOIN
+	dns_names D ON joinItem=2 AND E.id=D.registered_entry_id
+LEFT JOIN
+	(federated_registration_entries F INNER JOIN bundles B ON F.bundle_id=B.id) ON joinItem=3 AND E.id=F.registered_entry_id
+WHERE E.id IN (
+	SELECT id FROM (
+		SELECT id FROM registered_entries WHERE parent_id = ?
+		UNION
+		SELECT registered_entry_id AS id FROM selectors WHERE type = ? AND value = ?
+		UNION
+		SELECT registered_entry_id AS id FROM selectors WHERE type = ? AND value = ?
+	) listing
+)
+ORDER BY e_id, selector_id, dns_name_id
+;`,
+		},
+
+		{
+			dialect: "mysql",
+			by:      []string{"parent-id", "selector-exact-one"},
+			query: `
+SELECT
+	E.id AS e_id,
+	E.entry_id AS entry_id,
+	E.spiffe_id,
+	E.parent_id,
+	E.ttl AS reg_ttl,
+	E.admin,
+	E.downstream,
+	E.expiry,
+	S.id AS selector_id,
+	S.type AS selector_type,
+	S.value AS selector_value,
+	B.trust_domain,
+	D.id AS dns_name_id,
+	D.value AS dns_name
+FROM
+	registered_entries E
+LEFT JOIN
+	(SELECT 1 AS joinItem UNION SELECT 2 UNION SELECT 3) AS joinItems ON TRUE
+LEFT JOIN
+	selectors S ON joinItem=1 AND E.id=S.registered_entry_id
+LEFT JOIN
+	dns_names D ON joinItem=2 AND E.id=D.registered_entry_id
+LEFT JOIN
+	(federated_registration_entries F INNER JOIN bundles B ON F.bundle_id=B.id) ON joinItem=3 AND E.id=F.registered_entry_id
+WHERE E.id IN (
+	SELECT id FROM (
+		SELECT id FROM registered_entries WHERE parent_id = ?
+		UNION
+		SELECT registered_entry_id AS id FROM selectors WHERE type = ? AND value = ?
+	) listing
+)
+ORDER BY e_id, selector_id, dns_name_id
+;`,
+		},
+		{
+			dialect: "mysql",
+			by:      []string{"parent-id", "selector-exact-many"},
+			query: `
+SELECT
+	E.id AS e_id,
+	E.entry_id AS entry_id,
+	E.spiffe_id,
+	E.parent_id,
+	E.ttl AS reg_ttl,
+	E.admin,
+	E.downstream,
+	E.expiry,
+	S.id AS selector_id,
+	S.type AS selector_type,
+	S.value AS selector_value,
+	B.trust_domain,
+	D.id AS dns_name_id,
+	D.value AS dns_name
+FROM
+	registered_entries E
+LEFT JOIN
+	(SELECT 1 AS joinItem UNION SELECT 2 UNION SELECT 3) AS joinItems ON TRUE
+LEFT JOIN
+	selectors S ON joinItem=1 AND E.id=S.registered_entry_id
+LEFT JOIN
+	dns_names D ON joinItem=2 AND E.id=D.registered_entry_id
+LEFT JOIN
+	(federated_registration_entries F INNER JOIN bundles B ON F.bundle_id=B.id) ON joinItem=3 AND E.id=F.registered_entry_id
+WHERE E.id IN (
+	SELECT id FROM (
+		SELECT id FROM registered_entries WHERE parent_id = ?
+		UNION
```

Comment:
Shouldn't this be an intersection since we're filtering by both parent id and selectors?